tiledb/sm/query/writer.cc

```diff
@@ -1120,14 +1120,23 @@ Status Writer::filter_tiles(
         Status_WriterError("Incorrect number of tiles in filter_tiles"));
   }
 
+  // Reserve a vector for offsets tiles, they need to be processed after var
+  // data tiles as the processing of var data tiles depends on offset tiles.
+  std::vector<std::tuple<WriterTile*, WriterTile*, bool, bool>> args_offsets;
+  if (var_size) {
+    args_offsets.reserve(tile_num / tile_step);
+  }
+
   std::vector<std::tuple<WriterTile*, WriterTile*, bool, bool>> args;
-  args.reserve(tile_num);
+  args.reserve(tile_num - args_offsets.capacity());
```

Comment:
I am wondering if the capacity of a default initialized vector is guaranteed to be 0 as this code assumes in case of non var_size?

Reply:
I don't see any guarantees on capacity... So better not use that.